src/site/shared/tools/renderers/RotateToolRenderer.ts

```diff
@@ -0,0 +1,45 @@
+import {CirclePrim}       from "core/internal/view/rendering/prims/CirclePrim";
+import {CircleSectorPrim} from "core/internal/view/rendering/prims/CircleSectorPrim";
+
+import {ROTATION_CIRCLE_RADIUS, ROTATION_CIRCLE_THICKNESS, RotateTool} from "../RotateTool";
+
+import {ToolRenderer} from "./ToolRenderer";
+
+
+export const RotateToolRenderer: ToolRenderer<RotateTool | undefined> = {
+    isActive: (curTool): curTool is RotateTool | undefined => (!curTool || curTool instanceof RotateTool),
+
+    render: ({ circuit, renderer, curTool }) => {
+        const pos = circuit.selectionsMidpoint("world");
+
+        const drawOutline = () => {
+            renderer.draw(new CirclePrim(pos, ROTATION_CIRCLE_RADIUS, {
+                stroke: {
+                    color: "#ff0000",
+                    size:  ROTATION_CIRCLE_THICKNESS,
+                },
+                alpha: 0.5,
+            }));
+        }
+
+        const selections = circuit.selectedObjs;
+
+        // If we are in the default tool, draw the rotation circle outline if we have only components selected
+        if (!curTool) {
+            if (selections.length > 0 &&
+                selections.every((o) => (o.baseKind === "Component"))) {
+                drawOutline();
```

Comment:
This kind of indentation is kind of hard to read.  Usually I see this mitigated by putting the logical operators at the start of the line.  What's the policy on this?

Reply:
Yeah... I don't have a great solution here, I think I've been pretty consistent with having the logical operators at the ends of lines -- I think best policy may be to keep complex expressions out of if-statements generally anyways ? not sure

Reply:
> ...I think best policy may be to keep complex expressions out of if-statements generally anyways?...
I think this is a tough sell as a blanket policy, but I don't have a good solution either.

Reply:
It's kind of avoiding the problem, but i'd maybe factor out the lambda into a helper function somewhere so 29/30 fit on one line.  I imagine we want this kind of "base kind" filtering somewhere else too anyway ( :